### PR TITLE
EMP-356 bind mount qemu arm

### PR DESCRIPTION
**Pull request by oliv3r**
We merged this before the build could fail/succeed. This fixes our CI failure.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -30,21 +30,21 @@ cleanup()
 
 bootstrap_prepare()
 {
-    if [ ! -x "${ARM_EMU_BIN}" ]; then
+    if [ ! -x "${ARM_EMU_BIN}" ] || [ "${ARM_EMU_BIN#/}" != "${ARM_EMU_BIN}" ]; then
```
Comment (raysiudak) [inline build.sh]: In the alternate scenario.. when true, the binary is not checked for execution anymore. 
Maybe it is better to check for execution after the mount?

File: build_for_ultimaker.sh [hunk 2]
```diff
@@ -39,7 +39,7 @@ setup_emulation_support()
         fi
     done
 
-    if [ ! -x "${ARM_EMU_BIN}" ]; then
+    if [ ! -x "${ARM_EMU_BIN}" ] && [ "${ARM_EMU_BIN#/}" != "${ARM_EMU_BIN}" ]; then
```
Comment (raysiudak) [inline build_for_ultimaker.sh]: I am confused.. If you cannot execute the arm binary, and the arm binary is not equal to the arm binary without preceding slash.. then error. 
In this script the ARM_EMU_BIN should always have the full path including the starting '/'

File: build_for_ultimaker.sh [hunk 3]
```diff
@@ -39,7 +39,7 @@ setup_emulation_support()
         fi
```
Comment (raysiudak) [inline build_for_ultimaker.sh:38]: By the way, comment on your previous pull request:

grep interpreter "/proc/sys/fs/binfmt_misc/qemu-arm" | sed 's/interpreter //'

File: tests/buildenv.sh [hunk 4]
```diff
@@ -54,7 +54,7 @@ check_emulation_support()
         return
     fi
 
-    if [ ! -x "${ARM_EMU_BIN}" ]; then
+    if [ ! -x "${ARM_EMU_BIN}" ] || [ "${ARM_EMU_BIN#/}" != "${ARM_EMU_BIN}" ]; then
```
Comment (raysiudak) [inline tests/buildenv.sh]: Again it is very confusing what actually is going on here, what are your actually testing

File: tests/rootfs.sh [hunk 5]
```diff
@@ -12,7 +12,7 @@ RESULT=0
 
 setup()
 {
-    if [ ! -x "${ARM_EMU_BIN}" ]; then
+    if [ ! -x "${ARM_EMU_BIN}" ] || [ "${ARM_EMU_BIN#/}" != "${ARM_EMU_BIN}" ]; then
```
Comment (raysiudak) [inline tests/rootfs.sh]: Also it is nowhere stated in the usage of these script when and what to pass as env with the scripts

**Review by raysiudak — CHANGES_REQUESTED** [review]

File: build_for_ultimaker.sh (hunk 3 above)
Comment (oliv3r) [reply build_for_ultimaker.sh:38]: excellent, that's almost the one liner i was looking for and sed is part of busybox.
`
sed 's/interpreter //;t;d' /proc/sys/fs/binfmt_misc/qemu-arm
`
is the command i was looking for!

You are awesome ;)

File: build_for_ultimaker.sh (hunk 2 above)
Comment (oliv3r) [reply build_for_ultimaker.sh]: exactly, the ARM_EMU_BIN should always be an absolute path, thus always start with a /. This should be the default anyway.

The alternative is to add the / to the ROOTFS_DIR. which after sleeping over it, I think is better, albeit a bigger patch. Then we can support both absolute and relative paths.

**Review by raysiudak — APPROVED** [review]

**Pull request merged**